### ISSUE TITLE
Allow `optimize_weights` and `predict_probabilities` to take data frames as arguments

### ISSUE THREAD

**Issue by connormayer** [connormayer/maxent.ot#1]
Currently the `optimize_weights` and `predict_probabilities` functions take a path to an input/test file as an argument. It would be more flexible for users if the input/test file could be a pre-loaded data frame as well.

We'll probably want to create a new function in `read-data.R` to handle processing/validation of the input data frame, since most of the processing steps in `load_data_otsoft` will still need to be applied (e.g., filling in the blank input rows, extracting sample size, etc).

**Comment by adelrtan** [comment]
I happened to need to feed a data frame into `optimize_weights` as part of the `monte_carlo_weights` function that I wrote. That data frame was basically the output object of `predict_probabilities` with the `Predicted Probability`, `Observed Probability` and `Error` columns removed. i.e. the blank input rows were already filled in. So I basically added some code to `optimize_weights` to support taking in a data frame as an input. It didn't seem to bloat this function too much, so I didn't add it to `read-data.R` instead.

What scenarios are you thinking of in which a data frame may be an input, and we need to fill in the blank input rows?
